### PR TITLE
Add `rpart.plot()` to tuning article

### LINKED ISSUE #174: Add partykit plot back in?
We removed the partykit plot from whichever decision tree article it was because it was broken 😢 but now it is fixed 😄 in tidymodels/parsnip#316! We need to use `repair_call()`:

``` r
library(parsnip)
library(rpart)
library(partykit)
#> Loading required package: grid
#> Loading required package: libcoin
#> Loading required package: mvtnorm

tree_spec <- 
  decision_tree(
    cost_complexity = 1e-10,
    tree_depth = 4
  ) %>% 
  set_engine("rpart") %>% 
  set_mode("classification")

rpart_fit <- rpart(Kyphosis ~ Age + Number + Start, data = kyphosis)

parsnip_fit <- tree_spec %>%
  fit(Kyphosis ~ Age + Number + Start, data = kyphosis)

as.party(repair_call(parsnip_fit, kyphosis)$fit)
#> 
#> Model formula:
#> Kyphosis ~ Age + Number + Start
#> 
#> Fitted party:
#> [1] root
#> |   [2] Start >= 8.5
#> |   |   [3] Start >= 14.5: absent (n = 29, err = 0.0%)
#> |   |   [4] Start < 14.5
#> |   |   |   [5] Age < 55: absent (n = 12, err = 0.0%)
#> |   |   |   [6] Age >= 55
#> |   |   |   |   [7] Age >= 111: absent (n = 14, err = 14.3%)
#> |   |   |   |   [8] Age < 111: present (n = 7, err = 42.9%)
#> |   [9] Start < 8.5: present (n = 19, err = 42.1%)
#> 
#> Number of inner nodes:    4
#> Number of terminal nodes: 5
```

<sup>Created on 2020-05-28 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Let's decide if using the `repair_calll()` function is appropriate for any of the `/start` articles (too esoteric?) or perhaps somewhere else, and add it back in somewhere to show how to use it.

### PR DESCRIPTION
**Pull request by juliasilge**
Closes #174 

This PR does some editing to `/start/tuning` to use the new `extract_` functions, update URLs, and probably most importantly to add in a visualization of the decision tree using `rpart.plot()`.

**Question:** can someone else run the code in this article and see if they also now get the new tuning result? Are we not setting a seed somewhere that we should be? I now get this tuning result, rather [than what we had originally](https://www.tidymodels.org/start/tuning/#tune-grid):

image

I know we chatted about this at one point but I don't think we ever figured out what is causing the difference. I switched out the article here to choose by accuracy instead of ROC so we still choose the `tree_depth = 4` model.

I also changed to use svglite, addressing #213 for `/start`, because the `rpart.plot()` did not look great with so much fuzzy text. This is _not_ a full re-render, though; it is just waiting there to be applied to the rest of the articles.

### CODE REVIEW

**Comment by juliasilge** [comment]
Check out the [preview here](https://deploy-preview-223--tidymodels-org.netlify.app/start/tuning/). Nice crisp SVGs!

**Comment by EmilHvitfeldt** [comment]
I ran with CRAN packages and got the same chart. There are plenty of seed calls so it must be something else.

Screen Shot 2021-08-16 at 10 46 52 AM

Would it be possible to add a note/link on what the variable importance is calculation in the VIP plot?
from https://koalaverse.github.io/vip/reference/vi_model.html I get that

> As stated in one of the rpart vignettes. A variable may appear in the tree many times, either as a primary or a surrogate variable. **An overall measure of variable importance is the sum of the goodness of split measures for each split for which it was the primary variable, plus "goodness" * (adjusted agreement) for all splits in which it was a surrogate.** Imagine two variables which were essentially duplicates of each other; if we did not count surrogates, they would split the importance with neither showing up as strongly as it should. See rpart for details.

**Comment by juliasilge** [comment]
@hfrick can you take a look at these edits to this article? I still don't understand why we are getting different results for rpart compared to before but I think we'll just press on with what I have here, unless you see some explanation.

File: content/start/tuning/index.Rmarkdown [hunk 1]
```diff
@@ -168,18 +168,18 @@ tree_res %>%
...
+  show_best("accuracy")
 ```
 
-We can also use the [`select_best()`](https://tidymodels.github.io/tune/reference/show_best.html) function to pull out the single set of hyperparameter values for our best decision tree model:
+We can also use the [`select_best()`](https://tune.tidymodels.org/reference/show_best.html) function to pull out the single set of hyperparameter values for our best decision tree model:
 
 ```{r select-best-tree, dependson="tree-res"}
 best_tree <- tree_res %>%
-  select_best("roc_auc")
+  select_best("accuracy")
```
Comment (hfrick) [inline content/start/tuning/index.Rmarkdown:182]: the sentence below still refers to AUC :point_down:

**Review by hfrick — APPROVED** [review]
l get the same graph/results as you and Emil :woman_shrugging: :thinking:

**Comment by juliasilge** [comment]
The last time I looked at this I checked on RStudio Cloud on R 4.0.3, which is the same version as what the last render of this page had, and I _did_ see this difference. rpart itself hasn't changed since 2019. This is almost certainly a change in one of our packages.

At the time, Max said:

> However, in light of the RNGkind stuff from today, I'm betting that is the issue. I'll have a tune PR for you to test with soon.

I don't think we ever followed up. Given that we have something online that is pretty out-of-date with our current package versions, I am going to merge this. I do think this would be good to follow up on, though, and figure out what we actually want to do (change to old behavior? at least understand what changed?). 

cc @topepo for when you are back

**Pull request merged**